src/d3a/models/strategy/infinite_bus.py

```diff
@@ -45,33 +44,41 @@ def __init__(self, energy_sell_rate=None, energy_rate_profile=None, energy_buy_r
 
     def _set_market_maker_rate(self):
         if self.energy_rate_profile is not None:
-            GlobalConfig.market_maker_rate = \
-                read_and_convert_identity_profile_to_float(self.energy_rate_profile)
-        elif isinstance(self.energy_rate, (int, float)):
+            GlobalConfig.market_maker_rate = read_arbitrary_profile(
+                InputProfileTypes.IDENTITY, self.energy_rate_profile)
+        elif self.energy_rate is not None:
             GlobalConfig.market_maker_rate = self.energy_rate
-        elif isinstance(self.energy_rate, (str, dict)):
-            GlobalConfig.market_maker_rate = \
-                read_arbitrary_profile(InputProfileTypes.IDENTITY, self.energy_rate)
         else:
-            GlobalConfig.market_maker_rate = \
-                ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE
+            GlobalConfig.market_maker_rate = (
+                ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE)
 
-    def event_activate(self, **kwargs):
+    def _populate_selling_rate(self):
         if self.energy_rate_profile is not None:
-            self.energy_rate = read_and_convert_identity_profile_to_float(self.energy_rate_profile)
+            self.energy_rate = read_arbitrary_profile(InputProfileTypes.IDENTITY,
+                                                      self.energy_rate_profile)
+            # TODO: to be checked via deleting in case increased memory is observed during runtime
```

Comment:
I will take care of it when adapting to this in this PR:
https://github.com/gridsingularity/d3a/pull/1152/files#diff-b181cf173660ff13072297b6d1be9ab11df92dca436269dbef7da3ea0a523909
(src/d3a/models/strategy/infinite_bus.py) Or did you experience a large memory usage?

Reply:
As discussed in other [thread](https://github.com/gridsingularity/d3a/pull/1167#discussion_r665194267), its my intuition

Reply:
IMO this comment is not very descriptive (now we know what it means, but in 2 weeks we will not remember). Leave it though, as I will refactor the profile handling in D3ASIM-3421 anyways.